wagtail_localize/views/submit_translations.py

```diff
@@ -0,0 +1,212 @@
+
+
+class SubmitTranslationView(SingleObjectMixin, TemplateView):
+    template_name = "wagtail_localize/admin/submit_translation.html"
+    title = __("Translate")
+
+    def get_title(self):
+        return self.title
+
+    def get_subtitle(self):
+        return str(self.object)
+
+    def get_form(self):
+        if self.request.method == 'POST':
+            return SubmitTranslationForm(self.object, self.request.POST)
+        else:
+            return SubmitTranslationForm(self.object)
+
+    def get_success_url(self):
+        return get_valid_next_url_from_request(self.request)
+
+    def get_default_success_url(self):
+        pass
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        context.update({
+            "form": self.get_form(),
+            "next_url": self.get_success_url(),
+            "back_url": self.get_success_url() or self.get_default_success_url(),
+        })
+        return context
+
+    def post(self, request, **kwargs):
+        form = self.get_form()
+
+        if form.is_valid():
+            with transaction.atomic():
+                translator = TranslationCreator(self.request.user, form.cleaned_data["locales"])
+                translator.create_translations(self.object)
+
+                # Now add the sub tree (if the obj is a page)
+                if isinstance(self.object, Page):
+                    if form.cleaned_data["include_subtree"]:
+                        def _walk(current_page):
+                            for child_page in current_page.get_children():
+                                translator.create_translations(child_page)
+
+                                if child_page.numchild:
+                                    _walk(child_page)
+
+                        _walk(self.object)
+
+                if len(form.cleaned_data["locales"]) == 1:
+                    locales = form.cleaned_data["locales"][0].get_display_name()
+
+                else:
+                    # Note: always plural
+                    locales = _('{} locales').format(len(form.cleaned_data["locales"]))
+
+                # TODO: Button that links to page in translations report when we have it
+                messages.success(
+                    self.request, self.get_success_message(locales)
+                )
+
+                return redirect(self.get_success_url() or self.get_default_success_url())
+
+        context = self.get_context_data(**kwargs)
+        return self.render_to_response(context)
+
+    def dispatch(self, request, *args, **kwargs):
+        if not request.user.has_perms(['wagtail_localize.submit_translation']):
+            raise PermissionDenied
+
+        self.object = self.get_object()
+        return super().dispatch(request, *args, **kwargs)
+
+
+class SubmitPageTranslationView(SubmitTranslationView):
+    title = __("Translate page")
+
+    def get_subtitle(self):
+        return self.object.get_admin_display_title()
```

Comment:
I dont recall but the admin display title falls back to the page title if it's not explicitly set, right?

Reply:
Yep, it does: https://github.com/wagtail/wagtail/blob/master/wagtail/core/models.py#L876